### PR TITLE
(SIMP-4654) Fix runlevel enforcement

### PR DESCRIPTION
**Pull request by trevor-vaughan**
The `level` property functions in the providers were never actually
hooked back into the type. This meant that systems wouldn't actually
ever properly switch runlevels when using the `runelvel` type.

SIMP-4654 #comment Fix runlevel native type

### CODE REVIEW

File: lib/puppet/functions/simplib/module_exist.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Determines if a module exists in the current environment
```
Comment (lnemsick-simp) [inline lib/puppet/functions/simplib/module_exist.rb]: why was this added to this pr?

File: lib/puppet/type/runlevel.rb [hunk 2]
```diff
@@ -29,16 +29,35 @@ def initialize(*args)
     end
   end
```
Comment (lnemsick-simp) [inline lib/puppet/type/runlevel.rb]: where is the test for these changes

File: lib/puppet/functions/simplib/module_exist.rb (hunk 1 above)
Comment (trevor-vaughan) [reply lib/puppet/functions/simplib/module_exist.rb]: It's also for SIMP-4654

File: lib/puppet/type/runlevel.rb (hunk 2 above)
Comment (trevor-vaughan) [reply lib/puppet/type/runlevel.rb]: Right here: https://github.com/simp/pupmod-simp-gdm/pull/34/files#diff-9253f3c9cbffa1500f41d4db2baa8664R29

**Comment by jeefberkey** [comment]
:clock4: https://gitlab.com/simp/pupmod-simp-simplib/pipelines/24403827
